src/MatrixFields/single_field_solver.jl

```diff
@@ -54,10 +55,9 @@ function single_field_solve_diag_matrix_row!(
     A::ColumnwiseBandMatrixField,
     b,
 )
-    Aⱼs = unzip_tuple_field_values(Fields.field_values(A.entries))
-    b_vals = Fields.field_values(b)
-    x_vals = Fields.field_values(x)
-    (A₀,) = Aⱼs
+    # Use fields, and not field values, here so that this
+    # operation is mask-aware.
+    A₀ = A.entries.:1
     @. x_vals = inv(A₀) ⊠ b_vals
```

Comment:
@charleskawczynski I think this is a bug now, x_vals is not defined yet

Reply:
nor b_vals?

Reply:
Yep